src/main/scala/Core/control.scala

```diff
@@ -39,6 +41,7 @@ class ControlDatapathIO(implicit val conf: FlexpretConfiguration) extends Bundle
 
   // outputs to datapath (control dependent)
   val next_pc_sel     = Output(Vec(conf.threads, UInt(NPC_WI.W)))
+  val next_pc_sel_csr_addr = Output(UInt(12.W))
```

Comment:
What does this signal do?

Reply:
It is for the exception/interrupt implementation. 

There are two scenarios where we need to read a CSR register into the PC:
1. When an exception/interrupt occurs and we want to jump to the handler. Then we load `PC <- CSR_EVEC`.
2. When we return from an exception/interrupt and we want to jump back to the original PC. Then we load `PC <- CSR_MEPC`. 

We could add two more "enums" to the `next_pc_sel` signal, like so:

```
// next PCs select
val NPC_WI = 3
val NPC_PCREG = 0.U(3.W)
val NPC_PLUS4 = 1.U(3.W)
val NPC_BRJMP = 2.U(3.W)
val NPC_CSR_EVEC   = 3.U(3.W)
val NPC_CSR_MEPC   = 4.U(3.W)
```

But I found a more scalable approach would be doing it like so:

```
// next PCs select
val NPC_WI = 2
val NPC_PCREG = 0.U(2.W)
val NPC_PLUS4 = 1.U(2.W)
val NPC_BRJMP = 2.U(2.W)
val NPC_CSR   = 3.U(2.W)
```

Where the `NPC_CSR` means "read from some CSR". And then adding a new signal called `next_pc_sel_csr_addr` which encodes which CSR to read from. 

This also in part explains the code we talked about here: https://github.com/pretis/flexpret/pull/82#discussion_r1500361646